src/PeopleSDKAdapter.js

```diff
@@ -0,0 +1,119 @@
+ * This adapter utilizes the Webex JS SDK to fetch data about a person.
+ *
+ * @export
+ * @class PeopleSDKAdapter
+ * @extends {PeopleAdapter}
+ */
+export default class PeopleSDKAdapter extends PeopleAdapter {
+  constructor(datasource) {
+    super(datasource);
+
+    this.getPersonObservables = {};
+  }
+
+  /**
+   * Returns a proper Enum key for a given PersonStatus value
+   *
+   * @param {string} status
+   * @returns {string} PersonStatus
+   * @memberof PeopleSDKAdapter
+   */
+  getStatus(status) {
+    return Object.keys(PersonStatus).find((key) => PersonStatus[key] === status);
+  }
+
+  /**
+   * Encodes the personID into base64 format and return the UUID
+   *
+   * @param {string} ID
+   * @returns {string}
+   * @memberof PeopleSDKAdapter
+   */
+  getPersonUUID(ID) {
+    return Buffer.from(ID, 'base64')
+      .toString()
+      .split('/')[4]; // ciscospark://us/PEOPLE/<UUID>
+  }
+
+  /**
+   * Fetches the person data from the sdk and returns in the shape required by adapter.
+   *
+   * @param {string} ID
+   * @returns {Person}
+   * @memberof PeopleSDKAdapter
+   */
+  async fetchPerson(ID) {
+    const {id, emails, displayName, firstName, lastName, avatar, orgId} = await this.datasource.people.get(ID);
+
+    return {
+      ID: id,
+      emails,
+      displayName,
+      firstName,
+      lastName,
+      avatar,
+      orgID: orgId,
+    };
+  }
+
+  /**
+   * Returns an observable that emits person data of the given ID.
+   *
+   * @param {string} ID ID of person to get.
+   * @returns {Observable.<Person>}
+   * @memberof PeopleSDKAdapter
+   */
+  getPerson(ID) {
+    if (!(ID in this.getPersonObservables)) {
+      const personObs = from(this.fetchPerson(ID)).pipe(
+        map((data) => data),
+        (error) => error
+      );
+      const presenceObs = from(this.datasource.internal.presence.subscribe(this.getPersonUUID(ID))).pipe(
+        map((data) => data.responses[0].status.status, (error) => error)
+      );
+
+      // Subscribe to our internal `Apheleia` server to listen for any status change
+      // And update the person object with status response from the subscription
+      const personStatusObs = presenceObs.pipe(
+        flatMap((status) => personObs.pipe(map((person) => ({...person, status: this.getStatus(status)}))))
+      );
+
+      // Listen to any status change for the given person ID
+      const eventObs = fromEvent(this.datasource.internal.mercury, USER_PRESENCE_UPDATE_EVENT).pipe(
+        filter((event) => event.data.subject === this.getPersonUUID(ID)),
+        map((event) => this.getStatus(event.data.status))
+      );
+
+      // Update the person status after each change emitted from the event
+      const UpdatedPersonObs = personObs.pipe(
+        flatMap((person) => eventObs.pipe(map((status) => ({...person, status}))))
+      );
+
+      // Teardown logic
+      const teardownObs = Observable.create(() => async () => {
```

Comment:
Creation callbacks should not be `async` because we can't guarantee that they will resolve or when they will resolve